### ISSUE TITLE
mat-form-field should support mat-radio, mat-radio-group and mat-checkbox

### ISSUE THREAD

**Issue by naveedahmed1** [angular/material2#13365]
#### Bug, feature request, or proposal:

Please add support for `mat-radio`, `mat-radio-group`  and `mat-checkbox` in `mat-form-field`.

#### What is the expected behavior?
`mat-radio` and `mat-radio-group` should work inside `mat-form-field`  so that forms contains such controls have same formatting and error handling.

#### What is the current behavior?
It throws

```
ERROR Error: mat-form-field must contain a MatFormFieldControl.
    at getMatFormFieldMissingControlError
```

#### What are the steps to reproduce?

```
<mat-form-field >
      <mat-radio-group [formControlName]="id">
        <mat-radio-button *ngFor="let option of options" [value]="option.id">{{option.text}}</mat-radio-button>
      </mat-radio-group>
    </mat-form-field>
```
#### What is the use-case or motivation for changing an existing behavior?
So that the forms containing mat-radio and mat-radio-group and other controls have consistent formatting and error handling.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

```
Angular 7.0.0-beta.7
Material 7.0.0-beta.2
```

**Comment by mjamin** [comment]
See #8662.

**Comment by jelbourn** [comment]
Duplicate of #8662

**Comment by Vaishak** [comment]
@mjamin I went through the other issue and I still get the same error.  What is the workaround for this? 

@naveedahmed1 did the solution on #8662 help you to find a solution. If yes please share it , cause I couldn't make much sense out it. An ELI5 would be awesome. Thanks in advance.
